extensions/blocks/social-previews/upgrade.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { compact, get } from 'lodash';
+import { Button } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import upgradeImageUrl from './upgrade-illustration.svg';
+import analytics from '../../../_inc/client/lib/analytics';
+import { isSimpleSite } from '../../shared/site-type-utils';
+import getSiteFragment from '../../shared/get-site-fragment';
+
+const SocialPreviewsUpgrade = function SocialPreviewsUpgrade( {
+	autosaveAndRedirect,
+	href,
+	trackViewEvent,
+} ) {
+	// Using the effect here so the tracking is only called once on component mount.
+	useEffect( trackViewEvent, [] );
+
+	return (
+		<div className="jetpack-social-previews__modal-upgrade">
+			<img
+				className="jetpack-social-previews__upgrade-illustration"
+				src={ upgradeImageUrl }
+				width="351"
+				height="264"
+				alt="" // The image is decorative.
+			/>
+			<div className="jetpack-social-previews__upgrade-description">
+				<h2 className="jetpack-social-previews__upgrade-heading">
+					{ __( 'Upgrade to a Business Plan to unlock the power of our SEO tools', 'jetpack' ) }
+				</h2>
+				<ul className="jetpack-social-previews__upgrade-feature-list">
+					<li>
+						{ __(
+							'Preview your site’s content as it will appear on Facebook, Twitter, and the WordPress.com Reader.',
+							'jetpack'
+						) }
+					</li>
+					<li>
+						{ __(
+							'Control how page titles will appear on Google search results and social networks.',
+							'jetpack'
+						) }
+					</li>
+					<li>
+						{ __(
+							'Customize your front page meta data to change how your site appears to search engines.',
+							'jetpack'
+						) }
+					</li>
+				</ul>
+				<Button
+					href={ href } // Only for server-side rendering, since onClick doesn't work there.
+					isPrimary
+					label={ __( 'Purchase a business plan to access social previews', 'jetpack' ) }
+					onClick={ autosaveAndRedirect }
+					target="_top"
```

Comment:
I assume this is required for SSR?

Reply:
Do you mean the `target` specifically? It makes the code compatible with Calypso because the editor could be loaded inside an iframe and links without specified target will also open in the same iframe. By using `_top`, we will always update the location of the whole top window, not just inside an iframe (which would be wrong here).